### PR TITLE
[#6913] Added a support for GRANT .. WITH GRANT OPTION

### PR DESCRIPTION
**Pull request by timur-sh**

### CODE REVIEW

**Comment by lukaseder** [comment]
Fantastic, thank you very much! I'll review and merge immediately!

File: jOOQ/src/main/java/org/jooq/GrantGrantedStep.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import static org.jooq.SQLDialect.MYSQL;
+import static org.jooq.SQLDialect.POSTGRES;
+
+/**
+ * The step of creation <code>GRANT</code> statement that assign user
+ * or role a grant option.
+ *
+ * @author Timur Shaidullin
+ */
+public interface GrantGrantedStep extends GrantFinalStep{
```
Comment (lukaseder) [inline jOOQ/src/main/java/org/jooq/GrantGrantedStep.java]: Usually, the naming scheme is indicating what's possible in the current step, not what has been done in the previous step(s) as that could be ambiguous. So, a good name would be `GrantWithGrantOptionStep` or `GrantGrantOptionStep`. I'll fix this after merging.

File: jOOQ/src/main/java/org/jooq/GrantGrantedStep.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @Support({ POSTGRES, MYSQL, HSQLDB, FIREBIRD })
```
Comment (lukaseder) [inline jOOQ/src/main/java/org/jooq/GrantGrantedStep.java]: Just to comment, I'll remove the individual dialect references until we cover the functionality in our integration tests. While it is quite probable that this is correct, we want to *formally* support the feature by guaranteeing that it works...

Comment (timur-sh) [reply jOOQ/src/main/java/org/jooq/GrantGrantedStep.java]: Do you plan make an integration test open in the future? This step a bit decrease your workload, if users write integration test by themselves.

File: jOOQ/src/main/java/org/jooq/GrantGrantedStep.java (hunk 1 above)
Comment (timur-sh) [reply jOOQ/src/main/java/org/jooq/GrantGrantedStep.java]: Name `GrantGrantOptionStep` is more suitable for this case, it lets to add `[ GRANTED BY <grantor> ]` in the future.

File: jOOQ/src/main/java/org/jooq/GrantGrantedStep.java (hunk 2 above)
Comment (lukaseder) [reply jOOQ/src/main/java/org/jooq/GrantGrantedStep.java]: No, the integration tests are closed source and available to our paying customers only. We follow a similar dual-licensing business model as SQLite, which also does not open source all the tests.

File: jOOQ/src/main/java/org/jooq/GrantGrantedStep.java (hunk 1 above)
Comment (lukaseder) [reply jOOQ/src/main/java/org/jooq/GrantGrantedStep.java]: Yeah, perhaps, although I'd like to wait for someone to actually request the `GRANTED BY <grantor>` feature before implementing that...

**Pull request merged**